applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_geo_steady_state_piping_element.cpp

```diff
@@ -0,0 +1,293 @@
+{
+    // Arrange
+    const GeoSteadyStatePwPipingElement<2, 2> element;
+    const auto p_geometry   = std::make_shared<Line2D2<Node>>(CreateNodes());
+    const auto p_properties = std::make_shared<Properties>();
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementCanCreateInstanceWithNodeInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto nodes        = CreateNodes();
+    const auto p_properties = std::make_shared<Properties>();
+
+    // The source element needs to have a geometry, otherwise the version of the
+    // Create method with a node input will fail.
+    const auto p_geometry = std::make_shared<Line2D2<Node>>(CreateNodes());
+    const GeoSteadyStatePwPipingElement<2, 2> element(0, p_geometry, p_properties);
+
+    // Act
+    const auto p_created_element = element.Create(1, nodes, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsTheExpectedDoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_properties = std::make_shared<Properties>();
+
+    Model model;
+    const auto element = CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(model, p_properties);
+
+    // Act
+    const auto              dummy_process_info = ProcessInfo{};
+    Element::DofsVectorType degrees_of_freedom;
+    element.GetDofList(degrees_of_freedom, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 2);
+    for (auto dof : degrees_of_freedom) {
+        KRATOS_EXPECT_EQ(dof->GetVariable(), WATER_PRESSURE);
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsTheExpectedEquationIdVector,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_properties = std::make_shared<Properties>();
+
+    Model model;
+    auto element = CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(model, p_properties);
+
+    unsigned int i = 0;
+    for (const auto& node : element.GetGeometry()) {
+        ++i;
+        node.pGetDof(WATER_PRESSURE)->SetEquationId(i);
+    }
+
+    // Act
+    const auto                    dummy_process_info = ProcessInfo{};
+    Element::EquationIdVectorType equation_id_vector;
+    element.EquationIdVector(equation_id_vector, dummy_process_info);
+
+    // Assert
+    const std::vector<int> expected_ids = {1, 2};
```

Comment:
I would prefer to use `Element::EquationIdVectorType` rather than `std::vector<int>`, since then both vector types are identical. Now there is a difference in "signedness" of the integers, which I would prefer to avoid.
```suggestion
    const Element::EquationIdVectorType expected_ids = {1, 2};
```

Reply:
Yes that makes sense to me.
Done.